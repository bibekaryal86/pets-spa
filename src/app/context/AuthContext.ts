import {createContext} from 'react';
import {DefaultUserDetails, UserDetails,} from '../../home/types/home.data.types';

export interface AuthState {
    isLoggedIn: boolean;
    token: string;
    userDetails: UserDetails;
}

export interface AuthAction {
    authState: AuthState;
}

export const DefaultAuthState = {
    isLoggedIn: false,
    token: '',
    userDetails: DefaultUserDetails,
};

export const AuthContext = createContext({
    auth: DefaultAuthState,
    login: (auth: AuthState) => auth,
});
