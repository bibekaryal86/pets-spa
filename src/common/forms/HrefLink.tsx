import { useCallback } from 'react';
import styled, { css } from 'styled-components';

interface HrefLinkProps {
  id: string;
  linkTo: string;
  title: string;
  disabled?: boolean;
  underline?: boolean;
  color?: string;
  margin?: string;

  onClick?(event: React.MouseEvent): void;
}

interface HrefLinkWrapperProps {
  underline?: boolean;
  color?: string;
  disabled?: boolean;
  margin?: string;
}

const HrefLinkWrapper = styled.a.attrs({
  className: 'hreflink-wrapper',
})<HrefLinkWrapperProps>`
  text-decoration: ${(props) => (props.underline ? 'underline' : '')};
  color: ${(props) => (props.color ? props.color : '')};
  margin: ${(props) => (props.margin ? props.margin : '')};

  ${({ disabled }) =>
    disabled
      ? css`
          pointer-events: none;
          cursor: default;
          color: Gray;
        `
      : ``}
`;

const HrefLink: React.FC<HrefLinkProps> = (props) => {
  const onClick = useCallback(
    (event: React.MouseEvent) => {
      props.onClick && props.onClick(event);
    },
    [props],
  );

  return (
    <>
      <HrefLinkWrapper
        id={props.id}
        href={props.linkTo}
        onClick={onClick}
        color={props.color}
        underline={props.underline}
        disabled={props.disabled}
        margin={props.margin}
      >
        {props.title}
      </HrefLinkWrapper>
    </>
  );
};

export default HrefLink;
