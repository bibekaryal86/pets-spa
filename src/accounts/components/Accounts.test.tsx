/* eslint-disable @typescript-eslint/no-explicit-any */
import Accounts from './Accounts';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { AuthContext } from '../../app/context/AuthContext';
import { getAuthSample } from '../../common/fixtures/authContext.sample.data';
import * as accountsStateAction from '../actions/accounts.state.action';
import Select from '../../common/forms/Select';
import { ACCOUNTS_SAMPLE_DATA } from '../../common/fixtures/accounts.sample.data';
import {
  REF_ACCOUNT_TYPES,
  REF_ACCOUNT_TYPES_SELECT_OPTIONS,
  REF_BANKS,
  REF_BANKS_SELECT_OPTIONS,
  REF_STATUS_SELECT_OPTIONS,
} from '../../common/fixtures/refTypes.sample.data';
import {
  DisplayCardBody,
  DisplayCardRow,
  DisplayCardWrapper,
} from '../../styles/styled.card.style';
import HrefLink from '../../common/forms/HrefLink';
import Table from '../../common/forms/Table';

const mockUseNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockUseNavigate,
}));

describe('accounts tests', () => {
  const exampleUsername = 'example86';

  const getAccounts = jest.fn();
  const setAlert = jest.fn();
  const resetAlert = jest.fn();
  const resetOnPageLeave = jest.fn();
  const clearAccountsFilterSpy = jest.spyOn(
    accountsStateAction,
    'clearAccountsFilter',
  );
  const setAccountsFilterSpy = jest.spyOn(
    accountsStateAction,
    'setAccountsFilter',
  );

  const accountProps = {
    error: '',
    success: '',
    accountsList: [],
    accountTypes: REF_ACCOUNT_TYPES,
    banks: REF_BANKS,
    getAccounts,
    setAlert,
    resetAlert,
    resetOnPageLeave,
  };

  function renderAccounts(customProps = {}) {
    const props = { ...accountProps, ...customProps };

    return mount(
      <AuthContext.Provider value={getAuthSample()}>
        <Accounts {...props} />
      </AuthContext.Provider>,
    );
  }

  it('matches snapshot', () => {
    const output = renderAccounts();
    expect(toJson(output)).toMatchSnapshot();
  });

  describe('call to get accounts', () => {
    it('calls get accounts when accountsList is empty', () => {
      getAccounts.mockReset();
      renderAccounts();
      expect(getAccounts).toHaveBeenCalledWith(exampleUsername);
    });

    it('does not call get accounts when accountsList has value', () => {
      getAccounts.mockReset();
      renderAccounts({ accountsList: ACCOUNTS_SAMPLE_DATA });
      expect(getAccounts).not.toHaveBeenCalled();
    });
  });

  describe('alerts messages', () => {
    it('calls set alert when there is an error message', () => {
      renderAccounts({
        error: 'this-is-error',
      });
      expect(setAlert).toHaveBeenCalledWith('failure', 'this-is-error');
    });

    it('calls set alert when there is a success message', () => {
      renderAccounts({
        success: 'this-is-success',
      });
      expect(setAlert).toHaveBeenCalledWith('success', 'this-is-success');
      expect(clearAccountsFilterSpy).toHaveBeenCalled();
    });

    it('calls reset alert when leaving the page', () => {
      const output = renderAccounts();
      output.unmount();
      expect(resetAlert).toHaveBeenCalled();
      expect(resetOnPageLeave).toHaveBeenCalled();
    });
  });

  describe('accounts header', () => {
    const output = renderAccounts();

    it('header background color', () => {
      const headerBody = output.find(DisplayCardBody).at(0);
      expect(headerBody.prop('background')).toEqual('darkseagreen');
    });

    it('header text', () => {
      const headerText = output.find('h4');
      expect(headerText.text()).toEqual('Accounts');
    });
  });

  describe('account filters', () => {
    const output = renderAccounts({ accountsList: ACCOUNTS_SAMPLE_DATA });

    it('filters background color ', () => {
      const filtersBody = output.find(DisplayCardBody).at(1);
      expect(filtersBody.prop('background')).not.toBeDefined();
    });

    it('filters text', () => {
      const filtersBodyHeaderText = output.find(DisplayCardRow).at(0);
      expect(filtersBodyHeaderText.prop('fontWeight')).toEqual('bold');
      expect(filtersBodyHeaderText.text()).toEqual('Account Filters');
    });

    it('filters link to add new account', () => {
      const filterLinkText = output.find(DisplayCardRow).at(4);
      expect(filterLinkText.prop('borderTop')).toEqual(true);

      const hrefLink = filterLinkText.find(HrefLink);
      expect(hrefLink.prop('linkTo')).toEqual('#');
      expect(hrefLink.prop('title')).toEqual('To Add a New Account Click Here');

      const ahref = output.find('#accounts-add-new-account').at(0);
      expect(ahref.text()).toEqual('To Add a New Account Click Here');

      ahref.simulate('click');
      expect(mockUseNavigate).toHaveBeenCalledTimes(1);
      expect(mockUseNavigate).toHaveBeenCalledWith('/account/');
    });

    it('filter dropdowns', () => {
      const selects = output.find(Select);
      expect(selects.length).toEqual(3);

      expect(selects.at(0).prop('id')).toEqual('filter-account-select-type');
      expect(selects.at(0).prop('label')).toEqual('Account Type');
      expect(selects.at(1).prop('id')).toEqual('filter-account-select-bank');
      expect(selects.at(1).prop('label')).toEqual('Bank Name');
      expect(selects.at(2).prop('id')).toEqual('filter-account-select-status');
      expect(selects.at(2).prop('label')).toEqual('Account Status');
    });

    it('filter dropdown options', () => {
      const selects = output.find(Select);

      expect(selects.at(0).prop('options')).toEqual(
        REF_ACCOUNT_TYPES_SELECT_OPTIONS,
      );
      expect(selects.at(1).prop('options')).toEqual(REF_BANKS_SELECT_OPTIONS);
      expect(selects.at(2).prop('options')).toEqual(REF_STATUS_SELECT_OPTIONS);
    });

    it('filter dropdown option selected', () => {
      // select account type, dispatches to set account status as filter
      const statusSelect = output.find('#filter-account-select-status').at(1);
      statusSelect.simulate('change', {
        target: { value: 'ACTIVE' },
      });
      expect(setAccountsFilterSpy).toHaveBeenCalledTimes(1);
      expect(setAccountsFilterSpy).toHaveBeenCalledWith(
        'status',
        'ACTIVE',
        ACCOUNTS_SAMPLE_DATA,
      );
    });
  });

  describe('accounts filters applied', () => {
    const output = renderAccounts({ accountsList: ACCOUNTS_SAMPLE_DATA });
    let clearFiltersLink: any,
      displayCardWrappers: any,
      displayCardRowHeader: any;

    it('does not display selected filter', () => {
      expect(clearAccountsFilterSpy).toHaveBeenCalledTimes(1);

      displayCardWrappers = output.find(DisplayCardWrapper);
      expect(displayCardWrappers.length).toEqual(3);

      clearFiltersLink = output.find('#accounts-clear-filters');
      expect(clearFiltersLink.length).toEqual(0);
    });

    it('displays Account Status as selected filter', () => {
      const statusSelect = output.find('#filter-account-select-status').at(1);
      statusSelect.simulate('change', {
        target: { value: 'ACTIVE' },
      });

      displayCardWrappers = output.find(DisplayCardWrapper);
      expect(displayCardWrappers.length).toEqual(4);

      const displayCardRowHeader = displayCardWrappers
        .at(2)
        .find(DisplayCardRow)
        .at(0);
      expect(displayCardRowHeader.text()).toEqual(
        'Filters Currently Applied: [Account Status]',
      );
    });

    it('displays clear filter link', () => {
      clearFiltersLink = output.find('#accounts-clear-filters').at(1);
      expect(clearFiltersLink.text()).toEqual(
        'Clear Filters and Show All Accounts',
      );
    });

    it('displays Account Type as selected filter', () => {
      const typeSelect = output.find('#filter-account-select-type').at(1);
      typeSelect.simulate('change', {
        target: { value: '5ede4cf30525eb78290332e7' },
      });

      displayCardWrappers = output.find(DisplayCardWrapper);
      expect(displayCardWrappers.length).toEqual(4);

      displayCardRowHeader = displayCardWrappers
        .at(2)
        .find(DisplayCardRow)
        .at(0);
      expect(displayCardRowHeader.text()).toEqual(
        'Filters Currently Applied: [Account Type][Account Status]',
      );
    });

    it('displays Bank as selected filter', () => {
      const bankSelect = output.find('#filter-account-select-bank').at(1);
      bankSelect.simulate('change', {
        target: { value: '5ede4d8d0525eb78290332f0' },
      });

      displayCardWrappers = output.find(DisplayCardWrapper);
      expect(displayCardWrappers.length).toEqual(4);

      const displayCardRowHeader = displayCardWrappers
        .at(2)
        .find(DisplayCardRow)
        .at(0);
      expect(displayCardRowHeader.text()).toEqual(
        'Filters Currently Applied: [Account Type][Bank][Account Status]',
      );
    });

    it('clears filters', () => {
      clearFiltersLink.simulate('click');
      expect(clearAccountsFilterSpy).toHaveBeenCalledTimes(2);

      // does not display selected filters after clear
      displayCardWrappers = output.find(DisplayCardWrapper);
      expect(displayCardWrappers.length).toEqual(3);

      clearFiltersLink = output.find('#accounts-clear-filters');
      expect(clearFiltersLink.length).toEqual(0);
    });
  });

  describe('accounts list table', () => {
    const output = renderAccounts({ accountsList: ACCOUNTS_SAMPLE_DATA });

    describe('accounts list table contents', () => {
      const table = output.find(Table);

      it('accounts list table header', () => {
        const tableHeaders = table.find('thead').find('th');
        expect(tableHeaders.length).toEqual(5);

        const actualTableHeadersText = tableHeaders.map((tableHeader) =>
          tableHeader.text(),
        );
        const expectedTableHeadersText = [
          'Bank Name',
          'Account Type',
          'Account Name',
          'Opening Balance',
          'Current Balance',
        ];
        expect(actualTableHeadersText).toEqual(expectedTableHeadersText);
      });

      it('accounts list table footer', () => {
        const tableFooter = table.find('tfoot').find('tr').find('td');
        expect(tableFooter.text()).toEqual('Number of Records: 10');
      });

      describe('accounts list table body', () => {
        it('accounts list table body rows', () => {
          const tableBodyRows = table.find('tbody').find('tr');
          expect(tableBodyRows.length).toEqual(10);
        });

        it('accounts list table formatted positive balance', () => {
          const checkingAccountRow = table
            .find('tbody')
            .find('tr')
            .at(2)
            .find('td');
          expect(checkingAccountRow.length).toEqual(5);
          // length is same as `account list table header` test above
          const actualRowTexts = checkingAccountRow.map((tableRowText) =>
            tableRowText.text(),
          );
          const expectedRowTexts = [
            'BANK OF AMERICA',
            'CHECKING ACCOUNT',
            'BOFA-ADV PLUS',
            '$10,017.06',
            '$14,362.02',
          ];

          expect(actualRowTexts).toEqual(expectedRowTexts);
        });

        it('accounts list table formatted negative balance', () => {
          const creditCardRow = table.find('tbody').find('tr').at(4).find('td');
          expect(creditCardRow.length).toEqual(5);
          // length is same as `account list table header` test above
          const actualRowTexts = creditCardRow.map((tableRowText) =>
            tableRowText.text(),
          );
          const expectedRowTexts = [
            'CHASE BANK',
            'CREDIT CARD',
            'CHASE-AMAZON',
            '-$1.07',
            '-$14.05',
          ];

          expect(actualRowTexts).toEqual(expectedRowTexts);
        });

        it('accounts list click to open account details', () => {
          mockUseNavigate.mockReset();

          const accountNameColumn = table
            .find('tbody')
            .find('tr')
            .at(4)
            .find('td')
            .at(2);
          const accountNameLink = accountNameColumn.find(HrefLink);
          expect(accountNameLink.prop('linkTo')).toEqual('#');
          expect(accountNameLink.prop('title')).toEqual('CHASE-AMAZON');

          const accountNameLinkHref = accountNameLink.find('a');
          accountNameLinkHref.simulate('click');
          expect(mockUseNavigate).toHaveBeenCalledWith(
            '/account/5ede989a2c473171d7464592',
          );
        });
      });
    });
  });
});
