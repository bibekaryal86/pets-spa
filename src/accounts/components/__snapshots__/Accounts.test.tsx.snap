// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accounts tests matches snapshot 1`] = `
<Accounts
  accountTypes={
    Array [
      Object {
        "description": "ACCOUNT CASH",
        "id": "5ede4cbb0525eb78290332e4",
      },
      Object {
        "description": "CHECKING ACCOUNT",
        "id": "5ede4cc80525eb78290332e5",
      },
      Object {
        "description": "CREDIT CARD",
        "id": "5ede4cf30525eb78290332e7",
      },
      Object {
        "description": "INVESTMENT ACCOUNT",
        "id": "5fa83f9d465347404cc6aa21",
      },
      Object {
        "description": "LOANS AND MORTGAGES",
        "id": "5ede4d080525eb78290332e8",
      },
      Object {
        "description": "OTHER DEPOSIT ACCOUNTS",
        "id": "5ede4d170525eb78290332e9",
      },
      Object {
        "description": "OTHER LOAN ACCOUNTS",
        "id": "5ede4d1d0525eb78290332ea",
      },
      Object {
        "description": "SAVINGS ACCOUNT",
        "id": "5ede4cde0525eb78290332e6",
      },
    ]
  }
  accountsList={Array []}
  banks={
    Array [
      Object {
        "description": "AMERICAN EXPRESS",
        "id": "5ede4d5c0525eb78290332ec",
      },
      Object {
        "description": "BANK CASH",
        "id": "5ede4d510525eb78290332eb",
      },
      Object {
        "description": "BANK OF AMERICA",
        "id": "5ede4d680525eb78290332ed",
      },
      Object {
        "description": "CHASE BANK",
        "id": "5ede4d790525eb78290332ee",
      },
      Object {
        "description": "DISCOVER",
        "id": "5ede4d850525eb78290332ef",
      },
      Object {
        "description": "PNC BANK",
        "id": "5ede4d8d0525eb78290332f0",
      },
    ]
  }
  error=""
  getAccounts={
    [MockFunction] {
      "calls": Array [
        Array [
          "example86",
        ],
        Array [
          "example86",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  resetAlert={[MockFunction]}
  resetOnPageLeave={[MockFunction]}
  setAlert={[MockFunction]}
  success=""
>
  <styled.div
    id="accounts-body-header"
  >
    <div
      className="sc-fFGjHI gdyBHe display-card-wrapper"
      id="accounts-body-header"
    >
      <styled.div
        background="darkseagreen"
      >
        <div
          className="sc-dicizt cxzexS display-card-body"
        >
          <h4>
            Accounts
          </h4>
        </div>
      </styled.div>
    </div>
  </styled.div>
  <styled.div
    id="accounts-filter-content"
  >
    <div
      className="sc-fFGjHI gdyBHe display-card-wrapper"
      id="accounts-filter-content"
    >
      <styled.div>
        <div
          className="sc-dicizt gkQcEW display-card-body"
        >
          <styled.div
            borderBtm={true}
            fontWeight="bold"
          >
            <div
              className="sc-iVCKna fDINNh display-card-row"
              fontWeight="bold"
            >
              Account Filters
            </div>
          </styled.div>
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="four columns"
              >
                <styled.div>
                  <div
                    className="sc-iVCKna fSFIhA display-card-row"
                  >
                    <Select
                      className="u-full-width"
                      id="filter-account-select-type"
                      label="Account Type"
                      onChange={[Function]}
                      options={
                        Array [
                          Object {
                            "text": "Please Select",
                            "value": "",
                          },
                          Object {
                            "text": "ACCOUNT CASH",
                            "value": "5ede4cbb0525eb78290332e4",
                          },
                          Object {
                            "text": "CHECKING ACCOUNT",
                            "value": "5ede4cc80525eb78290332e5",
                          },
                          Object {
                            "text": "CREDIT CARD",
                            "value": "5ede4cf30525eb78290332e7",
                          },
                          Object {
                            "text": "INVESTMENT ACCOUNT",
                            "value": "5fa83f9d465347404cc6aa21",
                          },
                          Object {
                            "text": "LOANS AND MORTGAGES",
                            "value": "5ede4d080525eb78290332e8",
                          },
                          Object {
                            "text": "OTHER DEPOSIT ACCOUNTS",
                            "value": "5ede4d170525eb78290332e9",
                          },
                          Object {
                            "text": "OTHER LOAN ACCOUNTS",
                            "value": "5ede4d1d0525eb78290332ea",
                          },
                          Object {
                            "text": "SAVINGS ACCOUNT",
                            "value": "5ede4cde0525eb78290332e6",
                          },
                        ]
                      }
                      value=""
                    >
                      <styled.div
                        focus={false}
                      >
                        <div
                          className="sc-ewnqHT lfPEQV select-wrapper"
                        >
                          <styled.label
                            htmlFor="filter-account-select-type"
                          >
                            <label
                              className="sc-gTRrQi"
                              htmlFor="filter-account-select-type"
                            >
                              Account Type
                            </label>
                          </styled.label>
                          <select
                            className="u-full-width"
                            id="filter-account-select-type"
                            name="filter-account-select-type"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            size={0}
                            value=""
                          >
                            <Memo()
                              key=""
                              text="Please Select"
                              value=""
                            >
                              <option
                                hidden={false}
                                value=""
                              >
                                Please Select
                              </option>
                            </Memo()>
                            <Memo()
                              key="5ede4cbb0525eb78290332e4"
                              text="ACCOUNT CASH"
                              value="5ede4cbb0525eb78290332e4"
                            >
                              <option
                                hidden={false}
                                value="5ede4cbb0525eb78290332e4"
                              >
                                ACCOUNT CASH
                              </option>
                            </Memo()>
                            <Memo()
                              key="5ede4cc80525eb78290332e5"
                              text="CHECKING ACCOUNT"
                              value="5ede4cc80525eb78290332e5"
                            >
                              <option
                                hidden={false}
                                value="5ede4cc80525eb78290332e5"
                              >
                                CHECKING ACCOUNT
                              </option>
                            </Memo()>
                            <Memo()
                              key="5ede4cf30525eb78290332e7"
                              text="CREDIT CARD"
                              value="5ede4cf30525eb78290332e7"
                            >
                              <option
                                hidden={false}
                                value="5ede4cf30525eb78290332e7"
                              >
                                CREDIT CARD
                              </option>
                            </Memo()>
                            <Memo()
                              key="5fa83f9d465347404cc6aa21"
                              text="INVESTMENT ACCOUNT"
                              value="5fa83f9d465347404cc6aa21"
                            >
                              <option
                                hidden={false}
                                value="5fa83f9d465347404cc6aa21"
                              >
                                INVESTMENT ACCOUNT
                              </option>
                            </Memo()>
                            <Memo()
                              key="5ede4d080525eb78290332e8"
                              text="LOANS AND MORTGAGES"
                              value="5ede4d080525eb78290332e8"
                            >
                              <option
                                hidden={false}
                                value="5ede4d080525eb78290332e8"
                              >
                                LOANS AND MORTGAGES
                              </option>
                            </Memo()>
                            <Memo()
                              key="5ede4d170525eb78290332e9"
                              text="OTHER DEPOSIT ACCOUNTS"
                              value="5ede4d170525eb78290332e9"
                            >
                              <option
                                hidden={false}
                                value="5ede4d170525eb78290332e9"
                              >
                                OTHER DEPOSIT ACCOUNTS
                              </option>
                            </Memo()>
                            <Memo()
                              key="5ede4d1d0525eb78290332ea"
                              text="OTHER LOAN ACCOUNTS"
                              value="5ede4d1d0525eb78290332ea"
                            >
                              <option
                                hidden={false}
                                value="5ede4d1d0525eb78290332ea"
                              >
                                OTHER LOAN ACCOUNTS
                              </option>
                            </Memo()>
                            <Memo()
                              key="5ede4cde0525eb78290332e6"
                              text="SAVINGS ACCOUNT"
                              value="5ede4cde0525eb78290332e6"
                            >
                              <option
                                hidden={false}
                                value="5ede4cde0525eb78290332e6"
                              >
                                SAVINGS ACCOUNT
                              </option>
                            </Memo()>
                          </select>
                        </div>
                      </styled.div>
                    </Select>
                  </div>
                </styled.div>
              </div>
              <div
                className="four columns"
              >
                <styled.div>
                  <div
                    className="sc-iVCKna fSFIhA display-card-row"
                  >
                    <Select
                      className="u-full-width"
                      id="filter-account-select-bank"
                      label="Bank Name"
                      onChange={[Function]}
                      options={
                        Array [
                          Object {
                            "text": "Please Select",
                            "value": "",
                          },
                          Object {
                            "text": "AMERICAN EXPRESS",
                            "value": "5ede4d5c0525eb78290332ec",
                          },
                          Object {
                            "text": "BANK CASH",
                            "value": "5ede4d510525eb78290332eb",
                          },
                          Object {
                            "text": "BANK OF AMERICA",
                            "value": "5ede4d680525eb78290332ed",
                          },
                          Object {
                            "text": "CHASE BANK",
                            "value": "5ede4d790525eb78290332ee",
                          },
                          Object {
                            "text": "DISCOVER",
                            "value": "5ede4d850525eb78290332ef",
                          },
                          Object {
                            "text": "PNC BANK",
                            "value": "5ede4d8d0525eb78290332f0",
                          },
                        ]
                      }
                      value=""
                    >
                      <styled.div
                        focus={false}
                      >
                        <div
                          className="sc-ewnqHT lfPEQV select-wrapper"
                        >
                          <styled.label
                            htmlFor="filter-account-select-bank"
                          >
                            <label
                              className="sc-gTRrQi"
                              htmlFor="filter-account-select-bank"
                            >
                              Bank Name
                            </label>
                          </styled.label>
                          <select
                            className="u-full-width"
                            id="filter-account-select-bank"
                            name="filter-account-select-bank"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            size={0}
                            value=""
                          >
                            <Memo()
                              key=""
                              text="Please Select"
                              value=""
                            >
                              <option
                                hidden={false}
                                value=""
                              >
                                Please Select
                              </option>
                            </Memo()>
                            <Memo()
                              key="5ede4d5c0525eb78290332ec"
                              text="AMERICAN EXPRESS"
                              value="5ede4d5c0525eb78290332ec"
                            >
                              <option
                                hidden={false}
                                value="5ede4d5c0525eb78290332ec"
                              >
                                AMERICAN EXPRESS
                              </option>
                            </Memo()>
                            <Memo()
                              key="5ede4d510525eb78290332eb"
                              text="BANK CASH"
                              value="5ede4d510525eb78290332eb"
                            >
                              <option
                                hidden={false}
                                value="5ede4d510525eb78290332eb"
                              >
                                BANK CASH
                              </option>
                            </Memo()>
                            <Memo()
                              key="5ede4d680525eb78290332ed"
                              text="BANK OF AMERICA"
                              value="5ede4d680525eb78290332ed"
                            >
                              <option
                                hidden={false}
                                value="5ede4d680525eb78290332ed"
                              >
                                BANK OF AMERICA
                              </option>
                            </Memo()>
                            <Memo()
                              key="5ede4d790525eb78290332ee"
                              text="CHASE BANK"
                              value="5ede4d790525eb78290332ee"
                            >
                              <option
                                hidden={false}
                                value="5ede4d790525eb78290332ee"
                              >
                                CHASE BANK
                              </option>
                            </Memo()>
                            <Memo()
                              key="5ede4d850525eb78290332ef"
                              text="DISCOVER"
                              value="5ede4d850525eb78290332ef"
                            >
                              <option
                                hidden={false}
                                value="5ede4d850525eb78290332ef"
                              >
                                DISCOVER
                              </option>
                            </Memo()>
                            <Memo()
                              key="5ede4d8d0525eb78290332f0"
                              text="PNC BANK"
                              value="5ede4d8d0525eb78290332f0"
                            >
                              <option
                                hidden={false}
                                value="5ede4d8d0525eb78290332f0"
                              >
                                PNC BANK
                              </option>
                            </Memo()>
                          </select>
                        </div>
                      </styled.div>
                    </Select>
                  </div>
                </styled.div>
              </div>
              <div
                className="four columns"
              >
                <styled.div>
                  <div
                    className="sc-iVCKna fSFIhA display-card-row"
                  >
                    <Select
                      className="u-full-width"
                      id="filter-account-select-status"
                      label="Account Status"
                      onChange={[Function]}
                      options={
                        Array [
                          Object {
                            "text": "Please Select",
                            "value": "",
                          },
                          Object {
                            "text": "ACTIVE",
                            "value": "ACTIVE",
                          },
                          Object {
                            "text": "CLOSED",
                            "value": "CLOSED",
                          },
                        ]
                      }
                      value=""
                    >
                      <styled.div
                        focus={false}
                      >
                        <div
                          className="sc-ewnqHT lfPEQV select-wrapper"
                        >
                          <styled.label
                            htmlFor="filter-account-select-status"
                          >
                            <label
                              className="sc-gTRrQi"
                              htmlFor="filter-account-select-status"
                            >
                              Account Status
                            </label>
                          </styled.label>
                          <select
                            className="u-full-width"
                            id="filter-account-select-status"
                            name="filter-account-select-status"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            size={0}
                            value=""
                          >
                            <Memo()
                              key=""
                              text="Please Select"
                              value=""
                            >
                              <option
                                hidden={false}
                                value=""
                              >
                                Please Select
                              </option>
                            </Memo()>
                            <Memo()
                              key="ACTIVE"
                              text="ACTIVE"
                              value="ACTIVE"
                            >
                              <option
                                hidden={false}
                                value="ACTIVE"
                              >
                                ACTIVE
                              </option>
                            </Memo()>
                            <Memo()
                              key="CLOSED"
                              text="CLOSED"
                              value="CLOSED"
                            >
                              <option
                                hidden={false}
                                value="CLOSED"
                              >
                                CLOSED
                              </option>
                            </Memo()>
                          </select>
                        </div>
                      </styled.div>
                    </Select>
                  </div>
                </styled.div>
              </div>
            </div>
          </div>
          <styled.div
            borderTop={true}
          >
            <div
              className="sc-iVCKna cPlXiL display-card-row"
            >
              <HrefLink
                id="accounts-add-new-account"
                linkTo="#"
                onClick={[Function]}
                title="To Add a New Account Click Here"
              >
                <styled.a
                  href="#"
                  id="accounts-add-new-account"
                  onClick={[Function]}
                >
                  <a
                    className="sc-bgqQcB hQENut hreflink-wrapper"
                    href="#"
                    id="accounts-add-new-account"
                    onClick={[Function]}
                  >
                    To Add a New Account Click Here
                  </a>
                </styled.a>
              </HrefLink>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
  <Table
    data={Array []}
    exportToCsvFileName="Selected_Accounts.csv"
    footer="Number of Records: 0"
    headers={
      Array [
        "Bank Name",
        "Account Type",
        "Account Name",
        "Opening Balance",
        "Current Balance",
      ]
    }
    isExportToCsv={true}
    isSortAllowed={true}
    title="List of Accounts"
  >
    <styled.div>
      <div
        className="sc-fFGjHI gdyBHe display-card-wrapper"
      >
        <styled.div>
          <div
            className="sc-dicizt gkQcEW display-card-body"
          >
            <styled.div
              fontWeight="bold"
            >
              <div
                className="sc-iVCKna lhiroO display-card-row"
                fontWeight="bold"
              >
                List of Accounts
              </div>
            </styled.div>
            <styled.div>
              <div
                className="sc-iVCKna fSFIhA display-card-row"
              >
                <styled.div>
                  <div
                    className="sc-gxYJeL pznue table-max-width-wrapper"
                  >
                    <styled.div>
                      <div
                        className="sc-kPTPQs lhLxbL table-scroll-wrapper"
                      >
                        <styled.table>
                          <table
                            className="sc-ispOId bmqYLd table-wrapper"
                          >
                            <styled.thead>
                              <thead
                                className="sc-kcuKUB kJmmRa"
                              >
                                <styled.tr>
                                  <tr
                                    className="sc-duSInm lBjcF"
                                  >
                                    <styled.th
                                      isSortAllowed={true}
                                      key="0"
                                      onClick={[Function]}
                                    >
                                      <th
                                        className="sc-jrrXlR kzDgxC"
                                        onClick={[Function]}
                                      >
                                        Bank Name
                                      </th>
                                    </styled.th>
                                    <styled.th
                                      isSortAllowed={true}
                                      key="1"
                                      onClick={[Function]}
                                    >
                                      <th
                                        className="sc-jrrXlR kzDgxC"
                                        onClick={[Function]}
                                      >
                                        Account Type
                                      </th>
                                    </styled.th>
                                    <styled.th
                                      isSortAllowed={true}
                                      key="2"
                                      onClick={[Function]}
                                    >
                                      <th
                                        className="sc-jrrXlR kzDgxC"
                                        onClick={[Function]}
                                      >
                                        Account Name
                                      </th>
                                    </styled.th>
                                    <styled.th
                                      isSortAllowed={true}
                                      key="3"
                                      onClick={[Function]}
                                    >
                                      <th
                                        className="sc-jrrXlR kzDgxC"
                                        onClick={[Function]}
                                      >
                                        Opening Balance
                                      </th>
                                    </styled.th>
                                    <styled.th
                                      isSortAllowed={true}
                                      key="4"
                                      onClick={[Function]}
                                    >
                                      <th
                                        className="sc-jrrXlR kzDgxC"
                                        onClick={[Function]}
                                      >
                                        Current Balance
                                      </th>
                                    </styled.th>
                                  </tr>
                                </styled.tr>
                              </thead>
                            </styled.thead>
                            <styled.tbody>
                              <tbody
                                className="sc-kuWgmH"
                              >
                                <styled.tr>
                                  <tr
                                    className="sc-duSInm lBjcF"
                                  >
                                    <styled.td
                                      colSpan={100}
                                    >
                                      <td
                                        className="sc-crozmw iSpOGA"
                                        colSpan={100}
                                      />
                                    </styled.td>
                                  </tr>
                                </styled.tr>
                              </tbody>
                            </styled.tbody>
                            <styled.tfoot>
                              <tfoot
                                className="sc-dcrmVg dWSptR"
                              >
                                <styled.tr>
                                  <tr
                                    className="sc-duSInm lBjcF"
                                  >
                                    <styled.td
                                      colSpan={100}
                                    >
                                      <td
                                        className="sc-crozmw iSpOGA"
                                        colSpan={100}
                                      >
                                        Number of Records: 0
                                      </td>
                                    </styled.td>
                                  </tr>
                                </styled.tr>
                              </tfoot>
                            </styled.tfoot>
                          </table>
                        </styled.table>
                      </div>
                    </styled.div>
                    0
                  </div>
                </styled.div>
              </div>
            </styled.div>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </Table>
</Accounts>
`;
